### DISCUSSION TITLE
Implementing OmniOutliner features

### DISCUSSION THREAD

**Discussion by danieltomasz** [vslinko/obsidian-outliner#108] [Ideas]
Omnioutliner has a slightly different approach  to  outlining than the Dynalist 

> 
> Move
> Use the following commands to move, indent, or outdent rows in your document:
> Move Up (Control-Command–↑)—Moves the selected row and its children up one row.
> 
> Move Down (Control-Command–↓)—Moves the selected row and its children down one row.
> 
> Move Left (Control-Command–←)—Moves the selected row and its children to the left while ignoring any of its siblings.
> 
> Move Right (Control-Command–→)—Moves the selected row and its children to the right, making them children of the preceding row.
> 
> Move Right/Left vs. Indent/Outdent
> When it comes to organizing the content of your outline, it’s useful to understand the differences between Moving a row right or left (up or down a level) and Indenting or Outdenting a row.
> 
> When you move a row right or left, it might seem like all you are doing is moving it and its children up or down a level in the hierarchy of your outline. However, Move Right/Left ignores any siblings (rows at the same level as the row you are moving), and will move right or left and up or down in the hierarchy as needed.
> 
> However, when you Indent or Outdent a row, that row takes its children along for the ride.
> 
> Indent (Command–])
> Indents the current row.
> Outdent (Command–[)
> Outdents the current row.
> Group (Option-Command–G)
> Indents the selected rows and groups them together under a new parent row.
> Ungroup (Option-Command–U)
> Ungroups the rows of a selected group. Ungrouping does not remove a parent row; you’ll need to delete that separately, if you so choose.
> 

(Move Up and Move down is already implemented in Outliner here, but either it doesn't work or I don't understand how it supposed to work , in any case it is working different than in Omnioutliner). Implementing Move Left and Move Left would be huge, grouping also would be nice too

Maybe those features are already exists, but I was not able to discover them. Implementing those options would solve one of my use case I disussed before  https://github.com/vslinko/obsidian-outliner/issues/79

OmniOutliner could be downloaded for free and tested for 2weeks
https://www.omnigroup.com/omnioutliner/

**Comment by vslinko** [comment]
I've tried OmniOutliner shortcuts:

- `Control-Command-Up/Down` works almost the same as `Command-Shift-Up/Down` in Outliner plugin
- `Control-Command-Left/Right` works almost the same as `Shift-Tab`/`Tab` in Outliner plugin
- There are no alternatives for `Command–[` `Command–]` `Option-Command–G` `Option-Command–U` in Outliner plugin

It's possible to make those alternatives, but I'm not sure that implementing `Command–[` `Command–]` is a good idea. They have kinda strange behaviour. Could you explain real use case for those?

![out](https://user-images.githubusercontent.com/205823/116463498-6be7f080-a873-11eb-9733-ce8487b22cf5.gif)

**Reply by danieltomasz** [reply]
Hi, I didn't have time to comment when I saw your response; Now I am again using Linux at work  on a daily basis so I cannot directly compare, but if I will have time I will try to make a gif or so

>`Control-Command-Up/Down` works almost the same as `Command-Shift-Up/Down` in Outliner plugin'

I think the biggest difference is you cannot move an item up one level up using `Command-Shift-Up` (Move list and sublist down), 
what you can do I you can move it up between its siblings and siblings of its parents, 
you cannot  move the item to its grandfather or any of sibling of the grandfather
I don't propose to replace the current behaviour but to add another shortcut that  will allow to that 


Another addition I would like to implement to the possibility to move item one level in tree structure while not moving it as a row (as an alternative to the standard already implemented behaviour as I described it here https://github.com/vslinko/obsidian-outliner/issues/79); so it will make an item the parent of its current right siblings (Discussing outline as the binary tree is easier, but I am not used to it, I skipped the discrete mathematics classes ;)


`Option-Command–G` `Option-Command–U` is an interesting way of working with group of items , but I will not use it that often I think so this isn't that crititical

When I will be back on mac and I will have time I will try to record gifs so maybe my use case will be more clear

On this level of granurality the subtle difference in behaviour may be unintuitive for someone, while natural for other persons
